packager/packager.cc

```diff
@@ -2024,6 +2082,62 @@ void validatePackage(core::Context ctx) {
         return;
     }
 
+    if (pkgInfo.isPreludePackage()) {
+        // We disallow `visible_to` annotations in prelude pakcages, as they're implicitly imported by all non-prelude
+        // packages.
+        for (auto &v : pkgInfo.visibleTo_) {
+            if (auto e = ctx.beginError(v.name.fullName.loc, core::errors::Packager::NoPreludeVisibleTo)) {
+                // TODO: an autocorrect to remove the `visible_to` declarations would be nice, but we would need to
+                // start tracking the decl loc.
+                e.setHeader("Prelude package `{}` may not include `{}` annotations", pkgInfo.name.toString(ctx),
+                            "visible_to");
+            }
+        }
+
+        for (auto &i : pkgInfo.importedPackageNames) {
+            auto &otherPkg = packageDB.getPackageInfo(i.name.mangledName);
+            if (!otherPkg.isPreludePackage()) {
+                if (auto e = ctx.beginError(i.loc, core::errors::Packager::PreludePackageImport)) {
+                    string_view import_;
+                    switch (i.type) {
+                        case core::packages::ImportType::Normal:
+                            import_ = "import";
+                            break;
+                        case core::packages::ImportType::TestUnit:
+                        case core::packages::ImportType::TestHelper:
+                            import_ = "test_import";
+                            break;
+                    }
+                    e.setHeader("Prelude package `{}` may not `{}` non-prelude package `{}`",
+                                pkgInfo.name.toString(ctx), import_, otherPkg.show(ctx));
+                }
+            }
+        }
+    } else {
+        // We only check implicit prelude imports for non-prelude packages.
```

Comment:
Why do prelude packages not have to be properly layered?

Reply:
Thanks for calling this out, I hadn't thought through the effects of prelude packages specifying a `layer` that isn't the lowest-level. I've added a test, and required that all prelude packages live in the lowest level.